src/app/services/courses.service.ts

```diff
@@ -43,47 +47,66 @@ const coursesList = [{
 
 @Injectable()
 export class CoursesService {
-  public getCourses () {
-    return coursesList;
+  private isPopupDisplayed;
+  private deleteCourseId;
+  private courses: BehaviorSubject<ICourse[]> = new BehaviorSubject([]);
+
+  constructor () {
+    this.isPopupDisplayed = false;
+  }
+
+  public get getCourses (): Observable<ICourse[]> {
+    return this.courses;
   }
 
-  public createNewCourse (course) {
+  public getCourseDeleay () {
```

Comment:
`getCourses` to jest bardziej uchwyt do streamu a `getCourseDeleay`  wysyła dane. Zrefaktoryzowałem nazwy tych metod aby były bardziej adekwatne do ich działań.